Flaky test: TestJSONFormatProgress

On Windows TP4 (https://github.com/docker/docker/pull/21106 ) https://jenkins.dockerproject.org/job/Docker-PRs-WoW-TP4/2485/console

```
09:53:39 ok  	github.com/docker/docker/pkg/stdcopy	0.035s	coverage: 100.0% of statements
09:53:39 --- FAIL: TestJSONFormatProgress (0.00s)
09:53:39 	streamformatter_test.go:88: ProgressMessage must be [=========================>                         ]     15 B/30 B 404913h53m30s, got: [=========================>                         ]     15 B/30 B 404913h53m29s
09:53:39 FAIL
09:53:39 coverage: 42.2% of statements
```

Difference is the 1 second at the end;

```
[=========================>                         ]     15 B/30 B 404913h53m30s
[=========================>                         ]     15 B/30 B 404913h53m29s
```

--- a/pkg/streamformatter/streamformatter_test.go
+++ b/pkg/streamformatter/streamformatter_test.go
@@ -4,6 +4,7 @@ import (
 	"encoding/json"
 	"errors"
 	"reflect"
+	"strings"
 	"testing"
 
 	"github.com/docker/docker/pkg/jsonmessage"
@@ -84,9 +85,19 @@ func TestJSONFormatProgress(t *testing.T) {
 	if msg.Status != "action" {
 		t.Fatalf("Status must be 'action', got: %s", msg.Status)
 	}
-	if msg.ProgressMessage != progress.String() {
-		t.Fatalf("ProgressMessage must be %s, got: %s", progress.String(), msg.ProgressMessage)
+
+	// The progress will always be in the format of:
+	// [=========================>                         ]     15 B/30 B 404933h7m11s
+	// The last entry '404933h7m11s' is the timeLeftBox.
+	// However, the timeLeftBox field may change as progress.String() depends on time.Now().
+	// Therefore, we have to strip the timeLeftBox from the strings to do the comparison.
+
+	// Compare the progress strings before the timeLeftBox
+	expectedProgress := "[=========================>                         ]     15 B/30 B"
+	if !strings.HasPrefix(msg.ProgressMessage, expectedProgress) {
+		t.Fatalf("ProgressMessage without the timeLeftBox must be %s, got: %s", expectedProgress, msg.ProgressMessage)
 	}
+
 	if !reflect.DeepEqual(msg.Progress, progress) {
 		t.Fatal("Original progress not equals progress from FormatProgress")
 	}

